### DISCUSSION TITLE
Regarding QGEMM contrib op definition

### DISCUSSION THREAD

**Discussion by rasagna-quic** [microsoft/onnxruntime#10815] [Other Q&A] [answered]
Hi,

This is regarding the definition of [QGEMM contrib operator](https://github.com/microsoft/onnxruntime/blob/master/docs/ContribOperators.md#com.microsoft.QGemm). I see that there is no beta parameter for this op where as it is present in the Gemm operator (Y = alpha A B + beta C). Even though there is no beta, the definition still uses beta for the computation of C scale as mentioned below.

> "Optional input tensor C. If not specified, the computation is done as if C is a scalar 0. The shape of C should be unidirectional broadcastable to (M, N). Its type is int32_t and must be quantized with zero_point = 0 and scale = alpha / beta * a_scale * b_scale."

Could someone please clarify the same?

**Comment by rasagna-quic** [comment, marked as answer]
I understood why beta is not required for qgemm, this question can be closed. 

Below is my understanding
A = sA * (qA - zA), where qA is quantized A, sA is scale and zA is zeropoint.
B = sB * (qB - zB)
C = sC * (qC - zC)
As per the definition zC=0 and sC = alpha/beta * sA * sB
C = alpha/beta * sA * sB * qC

Gemm = alpha *A*B + beta*C
QGemm = alpha * sA * (qA - zA) * sB * (qB - zB) + beta * (alpha/beta * sA * sB * qC)
              = alpha * sA * (qA - zA) * sB * (qB - zB) + alpha * sA * sB * qC

**Reply by snnn** [reply]
@yufenglee  what do you think?

**Reply by yufenglee** [reply]
Yes, it is right. beta can be fused into the scale of C: (alpha/beta * sA * sB).
